Race condition taking place in the `security.py` test

|Wazuh repo branch|Component|
|---|---|
| master | Framework |

## Description

This problem was initially witnessed in the issue opened to update the current [`test_worker.py`](https://github.com/wazuh/wazuh/issues/9999) unit tests. When this test was updated, the checks were failing, but the failures were not related to anything modified in the file.

Locally, the tests were passing without any problem, thus it was decided to use [ the following image ](https://github.com/wazuh/wazuh-jenkins/tree/master/dockers/util/test_unit) to do further testing. Using this image, this error was replicated in the [branch](https://github.com/wazuh/wazuh/tree/feature/9999-add-worker-test) created for the issue managed above, but also in the [master](https://github.com/wazuh/wazuh/tree/master) branch. 

This problem could be related to a modification done in [this issue](https://github.com/wazuh/wazuh/issues/11556), as previously the RBAC database was created by importing the module, and currently, it is created by calling the `create_rbac_db()` function.

The aim of this issue is to investigate the possible reason behind this failure and put a solution to it. 

### Examples of the error encountered
```
_______________________ ERROR at setup of test_load_spec _______________________
/usr/local/lib/python3.9/site-packages/sqlalchemy/ext/declarative/clsregistry.py:294: in __call__
    x = eval(self.arg, globals(), self._dict)
<string>:1: in <module>
    ???
E   NameError: name 'Roles' is not defined
```
```
E           sqlalchemy.exc.InvalidRequestError: When initializing mapper mapped class Roles->roles, expression 'User' failed to locate a name ("name 'User' is not defined"). If this is a class name, consider adding this relationship() to the <class 'wazuh.rbac.orm.Roles'> class after both dependent classes have been defined.
```

The full trace can be found here: [roles_error.zip](https://github.com/wazuh/wazuh/files/7939968/roles_error.zip)


## Checks

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)
<!-- If changes are made to any of the following components, uncomment the corresponding line 
- [x] **Integration tests** without failures for API configuration (`/wazuh-qa/tests/integration/test_api/test_config/`)
- [x] **System tests** for agent enrollment process (`/wazuh-qa/tests/system/test_cluster/test_agent_enrollment`)
- [x] **System tests** for agent info sync process in cluster (`/wazuh-qa/tests/system/test_cluster/test_agent_info_sync`)
- [x] **System tests** for agent key polling (`/wazuh-qa/tests/system/test_cluster/test_agent_key_polling`)
- [x] **System tests** for JWT invalidation (`/wazuh-qa/tests/system/test_jwt_invalidation`)
-->

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

--- a/framework/wazuh/tests/test_security.py
+++ b/framework/wazuh/tests/test_security.py
@@ -72,6 +72,12 @@ def db_setup():
             with patch('shutil.chown'), patch('os.chmod'):
                 with patch('api.constants.SECURITY_PATH', new=test_data_path):
                     import wazuh.rbac.orm as orm
+                    # Invalidate in-memory database
+                    conn = orm._engine.connect()
+                    orm._Session().close()
+                    conn.invalidate()
+                    orm._engine.dispose()
+
                     reload(orm)
                     orm.create_rbac_db()
                     import wazuh.rbac.decorators as decorators

